### PR TITLE
[WIP] micro-benchmark draft

### PR DESCRIPTION
**Pull request by mindest**
**Description**: a simple prototype for operator-level micro-benchmarking

User needs to provide some necessary information:
- `model`: mini-model for test
- `model_init_params`: list of param names for initializing `model`
- `input_params`: list of list of input dim names, needed for generating inputs: `[[dim names for first input], ...]`
- `input_scales`: dict of input shapes: `{dim_name: dim_variation}`
- `job_configs`: using fp16 or fp32, pt or ort

output: a `.csv` file with kernel performance stat. under different input scales and configs

### CODE REVIEW

**Comment by mindest** [comment]
Below commit(s):
- Add preprocessing script to modify the original model file, such that `nn.functional` functions are replaced by our own wrapping modules, e.g. `F.relu` is wrapped with `WrappedRelu(nn.Module)`, this enables us to hook inputs more easily.

Todos/Known issues:
- inplace versions of these functions, if exist, are temporarily excluded
- some functions like `torch.repeat()`, `view()` are to be considered
- hook currently does not support keyword arguments, such info will be missing in the hook results (seems it will be supported in the near future: [link](https://github.com/pytorch/pytorch/issues/35643)
- (more to come)

**Comment by mindest** [comment]
A better alternative for the above commit:

The new commit
- enables extracting input information of operators in a given model, via monkey patching.
- adds a README (updating).
